File: RealTimeChatApp-BE/src/main/java/com/levanminh/realtimechatapp/notification/Notification.java
package com.levanminh.realtimechatapp.notification;

import com.levanminh.realtimechatapp.message.MessageType;
import lombok.*;

@Builder
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Notification {
    private String chatId;
    private String content;
    private String senderId;
    private String recipientId;
    private String chatName;
    private MessageType messageType;
    private NotificationType notificationType;
    private byte[] media;
}

File: RealTimeChatApp-FE/src/app/page/main/models/notification.ts
export interface Notification {
  chatId?: string;
  content?: string;
  senderId?: string;
  recipientId?: string;
  messageType?: 'TEXT' | 'IMAGE' | 'VIDEO' | 'AUDIO';
  notificationType?: 'SEEN' | 'MESSAGE' | 'IMAGE' | 'VIDEO' | 'AUDIO';
  chatName?: string;
  media?: Array<string>;
}
